From fd7f29912c623aec10268b2d98803837fd3a1de6 Mon Sep 17 00:00:00 2001
From: Richard Zhao <b20223@freescale.com>
Date: Thu, 5 Feb 2009 15:38:46 +0800
Subject: [PATCH] ENGR00104958 sgtl5000: set default mic vol

Set default mic volume to 0xf. So users can record using default settings.

Signed-off-by: Richard Zhao <b20223@freescale.com>
---
 sound/soc/codecs/sgtl5000.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sound/soc/codecs/sgtl5000.c b/sound/soc/codecs/sgtl5000.c
index 9e7fc09..cd881b7 100644
--- a/sound/soc/codecs/sgtl5000.c
+++ b/sound/soc/codecs/sgtl5000.c
@@ -836,6 +836,9 @@ static int sgtl5000_codec_io_probe(struct snd_soc_codec *codec,
 
 	reg = sgtl5000_read(codec, SGTL5000_CHIP_ANA_ADC_CTRL);
 	reg &= ~SGTL5000_ADC_VOL_M6DB;
+	reg &= ~(SGTL5000_ADC_VOL_LEFT_MASK | SGTL5000_ADC_VOL_RIGHT_MASK);
+	reg |= (0xf << SGTL5000_ADC_VOL_LEFT_SHIFT)
+	       | (0xf << SGTL5000_ADC_VOL_RIGHT_SHIFT);
 	sgtl5000_write(codec, SGTL5000_CHIP_ANA_ADC_CTRL, reg);
 
 	reg = SGTL5000_LINE_OUT_MUTE |
-- 
1.5.4.4

